File: VideoHosting.Api/VideoHosting.Data/Entities/Video.cs
namespace VideoHosting.Data.Entities;
public class Video
{
    public Guid Id { get; set; }

    public string Name { get; set; }

    public string Description { get; set; }
        
    public DateTime DayOfCreation { get; set; }

    public int Views { get; set; }

    public virtual User User { get; set; }

    public virtual List<Commentary> Commentaries { get; set; } = new();

    public virtual List<UserLike> Likes { get; set; } = new();

    public virtual List<UserDislike> Dislikes { get; set; } = new();

    public string PhotoPath { get; set; }

    public string VideoPath { get; set; }
    
    public void AddCommentary(Commentary commentary)
    {
        Commentaries.Add(commentary);
    }
    public void DeleteCommentary(Commentary commentary)
    {
        Commentaries.Remove(commentary);
    }
}

File: VideoHostingUI/src/app/Models/VideoModels/Video.ts
import { UUID } from 'crypto';

export class Video {
  constructor(
    public id: UUID,
    public userId: string,
    public name: string,
    public dayOfCreation: Date,
    public views: number,
    public videoPath: string,
    public photoPath: string,
    public likes: number,
    public dislikes: number,
    public description: string,
    public liked: boolean,
    public disliked: boolean,
  ) {}
}
